### DISCUSSION TITLE
How to log hyperparameters in Tensorboard HPARAMS tab?

### DISCUSSION THREAD

**Discussion by 00sapo** [Lightning-AI/pytorch-lightning#19876] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Hello,
from the documentation, it's not completely clear to me how to log hyperparameters in Tensorboard.

The documentation is at https://lightning.ai/docs/pytorch/stable/extensions/logging.html#logging-hyperparameters

It's not clear if and how `torch.utils.tensorboard.writer.SummaryWriter.add_hparams()` should be called. Moreover, I'm using `self.logger.log_hyperparams(self.hparams, {"hp/metric_1": 0, "hp/metric_2": 0})` as in the example, but all I get is this

![image](https://github.com/Lightning-AI/pytorch-lightning/assets/22996003/a4a81059-cc1c-4030-be19-5cbe22db6f1e)

I would expect _Start Value_ to be 0, at least. The HPARAMS tab instead shows the usual empty hyperparameters warning.

Can somebody show me a simple example for using the HPARAMS tab?

**Comment by adosar** [comment]
@00sapo I have opened a similar discussion [here](https://github.com/Lightning-AI/pytorch-lightning/discussions/19960). Do you get "No hparams data was found."?

**Reply by 00sapo** [reply]
Yep

**Comment by leSullivan** [comment]
Any solutions ?

**Comment by mazurkin** [comment]
https://github.com/Lightning-AI/pytorch-lightning/discussions/19960#discussioncomment-16105275

**Comment by MukundaKatta** [comment]
HPARAMS needs two calls. `log_hyperparams(self.hparams, {"hp/val_acc": 0.0})` registers the hyperparams and placeholder metrics, then `self.log("hp/val_acc", acc)` inside `validation_step` with the exact same `hp/...` name produces the scalar history TB needs. Two gotchas: names must match prefix-for-prefix (`hp/val_acc` in `log_hyperparams` is not the same as `val_acc` in `self.log`), and call `log_hyperparams` once (in `on_train_start`) not every step, otherwise you get duplicate HPARAMS entries.
